hydra-web/src/js/app.js

```diff
@@ -180,25 +199,66 @@ function(
             $("#loginToken")[0].value = "token";
             app.loginDialog = alertify.minimalDialog($('#loginForm')[0]);
         },
+        unsudo: function(username, token) {
+            var unsudoUrl = "/authentication/unsudo";
+            $.ajax({
+                type: 'POST',
+                url: unsudoUrl,
+                data: {
+                    username: username,
+                    token: token,
+                    sudo: Cookies.get("sudo"),
+                },
+            });
+            Cookies.set("sudo", "", {expires:0});
+            app.user.set("sudo", "");
+            app.sudoCb(false, true);
+        },
+
         sudo:function() {
+            var sudoUrl = app.authprefix() + "/authentication/sudo";
             var checked =  $("#sudoCheckbox").is(':checked');
+            var username = Cookies.get("username");
+            var token = Cookies.get("token");
             if (checked) {
-                var username = Cookies.get("username");
-                var token = Cookies.get("token");
-                var sudoToken = Cookies.get("sudo");
                 if (!username || !token) {
                     alertify.error("Please login first");
-                } else if (sudoToken) {
-                    app.user.set("sudo", sudoToken);
-                } else {
-                    $("#loginToken")[0].value = "sudo";
-                    $("#loginUsername")[0].value = username;
-                    app.loginDialog = alertify.minimalDialog($('#loginForm')[0]);
+                    return;
                 }
             } else {
-                Cookies.set("sudo", "", {expires:0});
-                app.user.set("sudo", "");
+                app.unsudo(username, token);
+                return;
             }
+            $.ajax({
+                type: 'POST',
+                url: sudoUrl,
+                data: {
+                    user: username,
+                    token: token
+                },
+                dataType: 'text',
+                success: function (response) {
+                    var sudoToken = response;
+                    if (!sudoToken) {
+                        alertify.error("Authentication error");
+                        sudoCb(false, false);
+                    } else {
+                        var message = "Are you ready to be working as sudo?";
```

Comment:
This message text may be better: "Please confirm that you want to enable sudo privileges"